azure-kusto-ingest/azure/kusto/ingest/ingest_client.py

```diff
@@ -108,3 +127,37 @@ def ingest_from_blob(self, blob_descriptor: BlobDescriptor, ingestion_properties
         content = ingestion_blob_info_json
         queue_client = queue_service.get_queue_client(queue=random_queue.object_name, message_encode_policy=TextBase64EncodePolicy())
         queue_client.send_message(content=content)
+
+    def _validate_endpoint_service_type(self):
+        if self._endpoint_service_type is None or not self._endpoint_service_type.strip():
+            self._endpoint_service_type = self._retrieve_service_type()
+
+        if self._EXPECTED_SERVICE_TYPE != self._endpoint_service_type:
+            has_endpoint = True
```

Comment:
I don't see my previous comment so adding a new one. 
You don't need this variable, just do it this way:
```
if not self._suggested_endpoint_uri:
	self._suggested_endpoint_uri = self._generate_endpoint_suggestion(self._connection_datasource)
	if not self._suggested_endpoint_uri:
		raise KustoInvalidEndpointError(self._EXPECTED_SERVICE_TYPE, self._endpoint_service_type)

raise KustoInvalidEndpointError(self._EXPECTED_SERVICE_TYPE, self._endpoint_service_type, self._suggested_endpoint_uri)
```

Reply:
I find my way more readable, but I know most people prefer your way because it's more elegant and some even find it more readable. I'll update this now.